ext/drmgr/drmgr.h

```diff
@@ -515,6 +515,79 @@ bool
 drmgr_unregister_opcode_instrumentation_event(drmgr_opcode_insertion_cb_t func,
                                               int opcode);
 
+DR_EXPORT
+/**
+ * Registers a callback function for the fifth instrumentation stage:
+ * post-instrumentation analysis and transformations on each
+ * basic block.  drmgr will call \p func as the fifth of five
+ * instrumentation stages for each dynamic application basic block.
+ * Post-instrumentation passes are allowed to insert both meta and
+ * non-meta instructions, and are primarily intended for debugging prior
+ * instrumentation passes.
+ *
+ * All instrumentation must follow the guidelines for
+ * #dr_register_bb_event().
+ *
+ * \return false if the given priority request cannot be satisfied
+ * (e.g., \p priority->before is already ordered after \p
+ * priority->after) or the given name is already taken.
+ *
+ * @param[in]  func        The callback to be called.
+ * @param[in]  priority    Specifies the relative ordering of the callback.
+ *                         Can be NULL, in which case a default priority is used.
+ */
+bool
+drmgr_register_bb_post_instru_event(drmgr_xform_cb_t func, drmgr_priority_t *priority);
+
+DR_EXPORT
+/**
+ * Unregisters a callback function for the fifth instrumentation stage.
+ * \return true if unregistration is successful and false if it is not
+ * (e.g., \p func was not registered).
+ *
+ * The recommendations for dr_unregister_bb_event() about when it
+ * is safe to unregister apply here as well.
+ */
+bool
+drmgr_unregister_bb_post_instru_event(drmgr_xform_cb_t func);
+
+DR_EXPORT
+/**
+ * Registers callbacks for all five instrumentation passes at once, with a \p
+ * user_data parameter passed among them all, enabling data sharing for all
+ * five of them.  See the documentation for drmgr_register_bb_app2app_event(),
+ * drmgr_register_bb_instrumentation_event(), drmgr_register_bb_instru2instru_event(), and
+ * drmgr_register_bb_post_instru_event() for further details of each pass. The
+ * aforementioned routines are identical to this with the exception of the extra \p
+ * user_data parameter, which is an OUT parameter to the \p app2app_func and passed in to
+ * the four subsequent callbacks. The \p priority param can be NULL, in which case a
+ * default priority is used.
+ */
+bool
```

Comment:
Do we need `drmgr_register_bb_instrumentation_all_events`? If the user wants the fifth event, maybe they can call `drmgr_register_bb_post_instru_event` separately on their own?